src/EnergyPlus/EconomicLifeCycleCost.cc

```diff
@@ -1545,16 +1548,21 @@ void ComputePresentValue(EnergyPlusData &state)
     // compute single present values based on real discount rates
     for (jYear = 1; jYear <= elcc->lengthStudyYears; ++jYear) {
         // NIST 155 D.2.1.1 - Single Present Value (SPV) formula
-        {
-            auto const SELECT_CASE_var(elcc->discountConvention);
-            if (SELECT_CASE_var == iDiscConv::BeginOfYear) {
-                effectiveYear = double(jYear) - 1.0;
-            } else if (SELECT_CASE_var == iDiscConv::MidYear) {
-                effectiveYear = double(jYear) - 0.5;
-            } else if (SELECT_CASE_var == iDiscConv::EndOfYear) {
-                effectiveYear = double(jYear);
-            } else {
-            }
+        switch (elcc->discountConvention) {
```

Comment:
This can be done using a table lookup rather than a `switch/case`.

```
constexpr std::array<Real64, DiscConv::Num> DiscConv2EffectiveYearAdjustment = {1.0, 0.5, 0.0};
effectiveYear = double(jYear) - DiscConv2EffectiveYearAdjustment[elcc->discountConvention]; 
```

Reply:
Thank you!